app/controllers/mixins/ems_common_angular.rb

```diff
@@ -149,6 +149,8 @@ def get_task_args(ems)
         [user, params[:default_password], endpoint_opts]
       when 'ManageIQ::Providers::Lenovo::PhysicalInfraManager'
         [user, password, params[:default_hostname], params[:default_api_port], "token", false]
+      when 'ManageIQ::Providers::Hawkular::MiddlewareManager'
+        [params[:default_hostname], params[:default_api_port], user, password, params[:default_security_protocol], params[:default_tls_ca_certs], true]
```

Comment:
As you've found [here](https://github.com/ManageIQ/manageiq-providers-hawkular/pull/100#issuecomment-342187357), please pass `params[:default_password]` instead of encrypted `password` here.

Reply:
Why are not encrypting password?

Reply:
@aljesusg it seems it's because we're calling the API with the password on input. Other providers do the decryption before the actual call (like [here](https://github.com/ManageIQ/manageiq-providers-amazon/blob/c01d14541f245519595512824639d0412510885e/app/models/manageiq/providers/amazon/manager_mixin.rb#L56)). We're not. That might be a neat enhancement, @abonas ;)

I also think the use of SSL and such won't change anything here, it's MIQ internal thing. So yes, we're using plaintext passwords.